Feature/basst314/client gameworld ui basics

superernie77 opened the pull request.

basst314 committed:
Added CORS (Cross-Origin Resource Sharing) for API-Urls

basst314 committed:
Step method returns current game world

basst314 committed:
Added routes to start/stop game to webserver

basst314 committed:
Added implementation of space-client debug component

basst314 committed:
fixed e2e test

basst314 committed:
fixed lint errors

basst314 committed:
cache node_modules folder for faster builds

basst314 committed:
reverse log, so that most recent activities show up on top

basst314 committed:
removed cached directory, be safe and install every dep on a fresh copy

basst314 committed:
removed reverse order for now, results in a bug on test server

basst314 committed:
use unshift to push to front

superernie77 committed:
Update README.md

superernie77 committed:
Merge branch 'master' into feature/basst314/client-gameworld-ui-basics

basst314 committed:
world has content member

basst314 committed:
world has content member

basst314 commented:
@superernie77  when i start space-server locally and call http://127.0.0.1:8080/api/start i get a StackOverflowError right away:

`2017-01-04 21:43:56.523 [qtp1093972936-21] WARN  org.eclipse.jetty.server.HttpChannel - //127.0.0.1:8080/api/start
java.lang.StackOverflowError: null
	at com.google.gson.reflect.TypeToken.equals(TypeToken.java:285)
	at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:940)
	at com.google.gson.Gson.getAdapter(Gson.java:399)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:56)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:125)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:243)`

can you reproduce it?

Commit:
bugfix StackOverflow Exception

Commit:
return messages for start/stop

Commit:
new player instance if requested by dummy dao

Commit:
Merge remote-tracking branch 'origin/space_api' into space_api

superernie77 committed:
Merge branch 'feature/basst314/client-gameworld-ui-basics' into space_api

basst314 committed:
update to angular2-webpack-starter v5.2.0 due to break in peer-dependencies

basst314 committed:
remove node 4/5 for travis

basst314 committed:
return World object as simple DTO

basst314 committed:
returned node 4/5

basst314 commented:
changed SpaceWorldController in this branch to return simple World objects instead of SpaceWorld- fixed the issue

basst314 committed:
update deps and config to latest master of angular2-webpack-starter

basst314 committed:
install node 6.x via nvm

basst314 committed:
fix linting errors;
explicitly state node version in travis ci build script
let stop() return an empty gameworld to prevent '404 not found' errors

basst314 committed:
fixed travis build (run 'ci' instead of single steps, refactoring from v5.2.0)

basst314 committed:
update build script

basst314 committed:
Merge branches 'feature/basst314/client-gameworld-ui-basics' and 'space_api' of https://github.com/basst314/space into feature/basst314/client-gameworld-ui-basics

basst314 committed:
Merge branches 'feature/basst314/client-gameworld-ui-basics' and 'space_api' of https://github.com/basst314/space into feature/basst314/client-gameworld-ui-basics

basst314 committed:
Merge pull request #15 from basst314/superernie77-patch-1

Update README.md
